### PR TITLE
Scalar markets views

### PR DESCRIPTION
**Pull request by KadenZipfel**
This PR includes:
- Scalar market creation flow (https://github.com/protofire/omen-exchange/issues/1193)
- Scalar market main view (not including buy/sell/pool/etc.)

**Notes**: 
- The list view of scalar markets is incomplete (should display current prediction, but may need a new design). I will create an issue for this later on.
- The scalar market `ask_question_step` form's numerical inputs behaving not so beautifully if you go to the next step then go back, i.e. adds a decimal, removes 0 inputs. I will also create an issue for this later on.

### CODE REVIEW

**Comment by KadenZipfel** [comment]
https://github.com/protofire/omen-exchange/pull/1253#issuecomment-723428297

**Comment by pimato** [comment]
The board between the navigation and the scale is missing, example market (http://localhost:3000/#/0x2b1dacc1ac0eb579d50f115397ad750546b3be6f):
Bildschirmfoto 2020-11-16 um 11 57 05

should look like:
Bildschirmfoto 2020-11-16 um 11 58 43

**Figma**:
[Design Specification](https://www.figma.com/file/R056tSYMjvfwaFidEBItYy/Widgets-V2?node-id=1209%3A141)

File: app/src/components/market/common/market_scale/index.tsx [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+const StartingPointBox = styled.div<{ xValue: number }>`
+  position: absolute;
+  padding: 12px;
+  border: 1px solid ${props => props.theme.scale.box};
+  border-radius: 4px;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  top: 44px;
```
Comment (pimato) [inline app/src/components/market/common/market_scale/index.tsx:91]: should be top: 36px;

File: app/src/components/market/common/market_scale/index.tsx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+      : `right: 25px`}
+  transform: translateX(-50%);
+  background: white;
+`
+
+const StartingPointTitle = styled.p`
+  font-size: 14px;
+  font-weight: 500;
+  color: ${props => props.theme.colors.textColorDarker};
+  margin-bottom: 6px;
```
Comment (pimato) [inline app/src/components/market/common/market_scale/index.tsx:104]: margin-bottom: 2px;

**Comment by pimato** [comment]
the predicted price box is out of bound for this market(http://localhost:3000/#/0x016ac8f16d8bf19d5c81832418b53aa1eb3b34c5)
Bildschirmfoto 2020-11-16 um 12 06 40

**Review by pimato — CHANGES_REQUESTED** [review]
see comments.

**Review by Mi-Lan — CHANGES_REQUESTED** [review]
Screenshot 2020-11-16 at 12 12 36
It seems that according to figma "Question"(title) should be in whitish letters and question in black letters instead of other way around

**Comment by pimato** [comment]
> Screenshot 2020-11-16 at 12 12 36
> It seems that according to figma "Question"(title) should be in whitish letters and question in black letters instead of other way around

good catch, just updated my designs:
Scalar

**Comment by Mi-Lan** [comment]
this "Set" button in set allowance component is according to old button design...should that get automatically updated with button pr?
Screenshot 2020-11-16 at 12 31 24

File: app/src/components/market/sections/market_buy/market_buy_container.tsx [hunk 3]
```diff
@@ -3,16 +3,22 @@ import React from 'react'
...
 
 const MarketBuyContainer: React.FC<Props> = (props: Props) => {
-  const { marketMakerData, switchMarketTab } = props
+  const { isScalar, marketMakerData, switchMarketTab } = props
 
-  return <MarketBuy marketMakerData={marketMakerData} switchMarketTab={switchMarketTab} />
+  if (isScalar) {
+    return <ScalarMarketBuy marketMakerData={marketMakerData} switchMarketTab={switchMarketTab} />
+  } else {
+    return <MarketBuy marketMakerData={marketMakerData} switchMarketTab={switchMarketTab} />
```
Comment (Mi-Lan) [inline app/src/components/market/sections/market_buy/market_buy_container.tsx:20]: maybe a suggestion for this is to make it one line for "if (cond) return component" so without curly braces and then to omit else after that and to just "return component" saving 2 lines or at least make else also one line ...

**Comment by KadenZipfel** [comment]
> this "Set" button in set allowance component is according to old button design...should that get automatically updated with button pr?
> Screenshot 2020-11-16 at 12 31 24

Yeah it should be updated when we merge into master

**Review by Mi-Lan — APPROVED** [review]
Looks good!

**Review by liaojikunwork — APPROVED** [review]

**Comment by pimato** [comment]
@KadenZipfel its picky but the borders are not aligned (seems like a few pixels are missing), currently:

top border on the navigation:
Bildschirmfoto 2020-11-17 um 11 22 15

bottom border on the bottom navigation buttons:
Bildschirmfoto 2020-11-17 um 11 22 15

should be like the border on the market details widget:
Bildschirmfoto 2020-11-17 um 11 22 22

**Review by pimato — CHANGES_REQUESTED** [review]
see comment

**Review by pimato — APPROVED** [review]

**Pull request merged**